R/Flood_CamML_App/app.R

```diff
@@ -288,85 +288,548 @@ ui <- material_page(
+        "model_score"   = 0.3, # model_prediction
+        "model_class"   = "No Flooding",#ifelse(model_prediction >= 0.5, "Flooding","No Flooding"),
+        "user_response" = button_response
+      )
+    }
+    
+    # Grab the data from the different cams
+    mirlo_data <- store_cam_data(cam_name = "Mirlo", 
+                                 model_prediction = mirlo_predict(), 
+                                 button_response = button_info_model1$mirlo_button_info)
+    northdock_data <- store_cam_data(cam_name = "NorthDock", 
+                                 model_prediction = northdock_predict(), 
+                                 button_response = button_info_model1$northdock_button_info)
+    southdock_data <- store_cam_data(cam_name = "SouthDock", 
+                                 model_prediction = southdock_predict(), 
+                                 button_response = button_info_model1$southdock_button_info)
+    southocracoke_data <- store_cam_data(cam_name = "SouthOcracoke", 
+                                 model_prediction = southocracoke_predict(), 
+                                 button_response = button_info_model1$southocracoke_button_info)
+    
+    
+    # Join them into one table
+    data <- bind_rows(mirlo_data, northdock_data, southdock_data, southocracoke_data)
+    
+    # Append data to google sheet
+    suppressMessages(googlesheets4::sheet_append(ss = sheets_ID,
+                                                 data = data))
+    
+  })
+  
+  #------------------- Submit button for model 2 -------------------
+  
+  observeEvent(input$submit2,{
+    
+    shinyalert(
+      inputId = "shinyalert2",
+      title = "Submit?",
+      text = "Are you ready to submit your answers?",
+      size = "s",
+      closeOnEsc = FALSE,
+      closeOnClickOutside = FALSE,
+      html = FALSE,
+      type = "warning",
+      showConfirmButton = TRUE,
+      showCancelButton = TRUE,
+      confirmButtonText = "Yes",
+      confirmButtonCol = "#AEDEF4",
+      cancelButtonText = "No",
+      timer = 0,
+      imageUrl = "",
+      animation = TRUE
+    )
+  })
+  
+  # Final submission for model 2 (tab 2)
+  observeEvent(input$shinyalert2 ==T,{
+    req(input$shinyalert2)
+    
+    updateActionButton(session = session,
+                       inputId = "submit2",
+                       label = "SUBMITTED!", 
+                       icon = icon("ok",lib = "glyphicon"))
+    
+    shinyjs::disable("submit2")
+    
+    # Data for unsupervised model submissions
+    data <- tibble(
+      "date" = c(mirlo_time_reactive_unsupervised()),
+      "location" = c("Mirlo"),
+      "filename" = c("mirlo_filename.jpg"), #change to whatever it ends up being
+      "model_type" = "unsupervised",
+      "model_score" = 0.3, # mirlo_predict_unsupervised()
+      "model_class" = "No Flooding", #ifelse(mirlo_predict_unsupervised() >= 0.5, "Flooding","No Flooding"),
+      "user_response" = button_info_model2$mirlo_button_info
+    )
```

Comment:
Still has the old format, likely could use the store_cam_data function with some more code to add suffixes for unsupervised models